Tabnine/src/main/java/com/tabnine/lifecycle/ForceRegistration.kt

```diff
@@ -0,0 +1,194 @@
+        if (state?.userName != null) {
+            val notification = Notification("force-registration-greeting", NOTIFICATION_ICON, NotificationType.INFORMATION)
+            notification.setContent("You are signed in as ${state.userName}")
+            notification.addAction(object: AnAction("Ok") {
+                override fun actionPerformed(e: AnActionEvent) {
+                    notification.expire()
+                }
+            })
+            Notifications.Bus.notify(notification)
+        }
+
+    }
+
+    private fun presentNotification() {
+        println("presenting notification")
+        val notification = Notification("force-signin-id", NOTIFICATION_ICON, NotificationType.INFORMATION)
+        notification.setContent("Please sign in to start using Tabnine")
+        notification.isImportant = true
+        notification.addAction(object: AnAction("Sign in") {
+            override fun actionPerformed(e: AnActionEvent) {
+                notification.expire()
+                openSigninPage()
+            }
+        })
+        Notifications.Bus.notify(notification)
+
+    }
+
+    private fun subscribeLogin() {
+        println("subscribing to login")
+        val bus = ApplicationManager.getApplication().messageBus.connect()
+        bus.subscribe(
+            BinaryStateChangeNotifier.STATE_CHANGED_TOPIC,
+            BinaryStateChangeNotifier { response ->
+                if (response.isLoggedIn == true) {
+                    bus.disconnect()
+                    presentGreeting(response)
+                }
+            }
+        )
+    }
+    private fun invoke(state: StateResponse? = null) {
+        println("invoke called")
+        if (!CapabilitiesService.getInstance().isCapabilityEnabled(Capability.FORCE_REGISTRATION)) {
+            println("force registration is not set so returning")
+            return
+        }
+        // ServiceManager.getService(BinaryStateService.class).getLastStateResponse();
+        val loggedIn = if (state != null) state.isLoggedIn else ServiceManager.getService(BinaryStateService::class.java).lastStateResponse?.isLoggedIn
+
+        val isNewInstallation = PluginInstalled.isNewInstallation
+        println("loggedIn = $loggedIn, isNewInstallation = $isNewInstallation" )
+        when {
+
+            isNewInstallation == true && loggedIn == false -> {
+//                subscribeLogin()
+                presentPopup()
+                presentNotification()
+
+            }
+            isNewInstallation == true && loggedIn == true-> {
+                presentGreeting(state ?: ServiceManager.getService(BinaryStateService::class.java).lastStateResponse)
+            }
+            isNewInstallation == false && loggedIn == false -> {
+//                subscribeLogin()
+                presentNotification()
+            }
+            isNewInstallation == false && loggedIn == true -> {
+                presentNotification()
+            }
```

Comment:
isn't this just `isNewInstallation == false`?

Reply:
no because you want the toggle state of the login